Tasks/AzureResourceGroupDeployment/models/DeployAzureRG.ts

```diff
@@ -43,22 +68,44 @@ export class AzureRGTaskParameters {
             }
             this.overrideParameters = tl.getInput("overrideParameters");
             this.enableDeploymentPrerequisites = tl.getInput("enableDeploymentPrerequisites");
-            this.machineGroupName = tl.getInput("machineGroupName");
+            this.deploymentGroupName = tl.getInput("deploymentGroupName");
             this.copyAzureVMTags = tl.getBoolInput("copyAzureVMTags");
-            this.vstsPATToken = tl.getInput("vstsPATToken");
+            var deploymentGroupEndpointName = tl.getInput("deploymentGroupEndpoint", false);
+            if(deploymentGroupEndpointName){
+                this.tokenCredentials = this.getVSTSPatToken(deploymentGroupEndpointName);
+            }
             this.outputVariable = tl.getInput("outputVariable");
             this.deploymentMode = tl.getInput("deploymentMode");
             this.credentials = this.getARMCredentials(connectedService);
-            this.machineGroupCollectionUrl = tl.getVariable("__mg__internal__collection__uri") 
-                                            || tl.getVariable('system.TeamFoundationCollectionUri');
-            this.machineGroupProjectName = tl.getVariable("__mg__internal__project__name") 
-                                            || tl.getVariable('system.teamProject');
+            this.deploymentGroupProjectName = tl.getInput("project");
         }
         catch (error) {
             throw (tl.loc("ARGD_ConstructorFailed", error.message));
         }
     }
 
+    private getVSTSPatToken(deploymentGroupEndpointName: string): TokenCredentials {
+        var endpointAuth = tl.getEndpointAuthorization(deploymentGroupEndpointName, true);
+        if (endpointAuth.scheme === 'Token') {
+            var hostUrl = tl.getEndpointUrl(deploymentGroupEndpointName, true);
+            var patToken: string = endpointAuth.parameters["apitoken"];
+            if (!Boolean(hostUrl) || typeof hostUrl.valueOf() !== 'string') {
+                throw new Error(tl.loc("DeploymentGroupEndpointUrlCannotBeEmpty"));
+            }
+
+            if (!Boolean(patToken) || typeof patToken.valueOf() !== 'string') {
+                throw new Error(tl.loc("DeploymentGroupEndpointPatTokenCannotBeEmpty"));
+            }
+            var credentials = new TokenCredentials(hostUrl, patToken);
+            return credentials;
+        }
+        else {
+            var msg = tl.loc("OnlyTokenAuthAllowed");
+            console.log(msg);
```

Comment:
you don't need to do console.log for that. TestFx will print loc_OnlyTokenAuthAllowed which can be used for validation.